client/pages/table.tsx

```diff
@@ -3,120 +3,98 @@
+    $team: String
+    $position: String
+    $player: String
+    $limit: Int
+    $offset: Int
+  ) {
+    getAllPlayers(
+      team: $team
+      position: $position
+      player: $player
+      limit: $limit
+      offset: $offset
+    ) {
+      total
+      players {
+        player
+        position
+        age
+        team
+        gamesPlayed
+        gamesStarted
+        minutesPlayed
+        fieldGoals
+        fieldGoalAttempts
+        fieldGoalPercentage
+        threePointFieldGoals
+        threePointFieldGoalAttempts
+        threePointFieldGoalPercentage
+        twoPointFieldGoals
+        twoPointFieldGoalAttempts
+        twoPointFieldGoalPercentage
+        effectiveFieldGoalPercentage
+        freeThrows
+        freeThrowAttempts
+        freeThrowPercentage
+        offensiveRebounds
+        defensiveRebounds
+        totalRebounds
+        assists
+        steals
+        blocks
+        turnovers
+        personalFouls
+        pointsPerGame
+      }
+    }
+  }
+`;
 
 const table = () => {
-  const [players, setPlayers] = useState<Player[]>([]);
-  const [nbaPlayer, setNbaPlayer] = useState("");
   const [page, setPage] = useState(0); // what page to start on
-  const [limit, setLimit] = useState(10); // how many entries you want per page
-  const [disableNextButton, setDisableNextButton] = useState(false);
-  const [disablePreviousButton, setDisablePreviousButton] = useState(true);
+  const [limit, setLimit] = useState(5); // how many entries you want per page
   const [playerCount, setPlayerCount] = useState(0);
   const [team, setTeam] = useState(""); // keep track of what team you want to find
   const [position, setPosition] = useState(""); // position dropdown data
-  console.log(position);
-
+  const [SearchPlayer, setSearchPlayer] = useState(""); // search nbaPlayer
+  const [players, setPlayers] = useState([]); // mapped data
+  
   const lastPage = Math.floor(playerCount / limit); // the last page
-  console.log(team);
-  // update URL
-  // const updateUrl = (newLimit: number, newPage: number) => {
-  //   const url = `${window.location.search}?pages=${newPage}&limit=${newLimit}`;
-  //   window.history.pushState({}, "", url);
-  // };
-
-  //fetch the table data when program starts
-  useEffect(() => {
-    fetchPlayerData(page, limit, nbaPlayer, team, position);
-  }, [page, team, position, nbaPlayer]);
-  //
-  const fetchPlayerData = async (
-    page: number,
-    limit: number,
-    nbaPlayer: string,
-    team: string,
-    position: string
-  ) => {
-    try {
-      const url = new URL("/getPlayers", "http://localhost:8000");
-
-      url.searchParams.append("page", page.toString());
-      url.searchParams.append("limit", limit.toString());
 
-      if (nbaPlayer) url.searchParams.append("nbaPlayer", nbaPlayer);
-      if (team) url.searchParams.append("team", team);
-      if (position) url.searchParams.append("position", position);
+  const { data } = useQuery(GET_PLAYERS_QUERY, {
```

Comment:
i believe we can also do

```suggestion
  const { data, loading, error } = useQuery(GET_PLAYERS_QUERY, {
```

lets add some logic where we show some error information if the method call failed. feel we can also do this in another pr, up to you

Reply:
added error and loading page